src/coreComponents/CMakeLists.txt

```diff
@@ -123,3 +123,29 @@ if( GEOS_ENABLE_TESTS )
   add_subdirectory( unitTests )
 endif( )
 
+########################################################################
+# the following adds a `unittests` CMake target such that running `$ make unittests test`
+# builds the unit tests before running them
+function(get_all_test_targets var)
+    set(targets)
+    get_all_test_targets_recursive(targets ${CMAKE_CURRENT_SOURCE_DIR})
+    set(${var} ${targets} PARENT_SCOPE)
+endfunction()
+
+macro(get_all_test_targets_recursive targets dir)
+    get_property(subdirectories DIRECTORY ${dir} PROPERTY SUBDIRECTORIES)
+    foreach(subdir ${subdirectories})
+        get_all_test_targets_recursive(${targets} ${subdir})
+    endforeach()
+
+    get_property(current_targets DIRECTORY ${dir} PROPERTY BUILDSYSTEM_TARGETS)
+    foreach(target ${current_targets})
+      if(${target} MATCHES "(test|example).+")
+        list(APPEND ${targets} ${target})
+      endif()
+    endforeach()
+endmacro()
+
+get_all_test_targets(unittests_targets)
+add_custom_target(unittests COMMAND ${CMAKE_CTEST_COMMAND} DEPENDS ${unittests_targets})
```

Comment:
Can you better explain what you're trying to achieve here? I'm not sure to get it.

Reply:
OK I better understand now.
Is it possible to add those comments into the `cmake` code so the direct intention is clear.
Also, would a name like `build_tests` (or `build_test` if we want to keep the symmetry with `test`) be clearer?

IIUC, the pattern is to walk through all the folders and find targets which look alike tests, based on naming conventions.
Would the best way to do this be to add a function `geos_add_test` which would call `blt_add_test` as we do now, but also add the test to the custom target `unittest`/`build_tests`.
This would make the registration explicit and safer?

Other than that, is it possible to retrieve the tests which are included in the `test` target?
If we could, then we would not have to look for those?